src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -341,7 +341,8 @@ static void setCondsAndInitialNotifyNgsiv1
                                             status,
                                             fiwareCorrelator,
                                             sub.notification.attributes,
-                                            sub.notification.blacklist);
+                                            sub.notification.blacklist,
+                                            V2);
```

Comment:
The function where this call is located is setCondsAndInitialNotifyNgsiv1(). Thus, maybe it should be `V1`?

In that case, this is a bit weird... it should fail with `V2` as it is now or maybe the condition is not properly captured by a test (it shoulld).

Reply:
The setCondsAndInitialNotifyNgsiv1 seems to be called only if in the subscription update request (v1/updateContextSubscription ) was present the subject. But that field shouldn't be only for NGSIv2? So even if we wrongly pass V2 above, this function is not called at all. This function shouldn't be used when we have a subscription coming from NGSIv1 (subUp.fromNgsiv1) ? 
So, to summarize:
 * mongoUpdateSubscription -> 
* setCondsAndInitialNotify (wrapper)
*  ->  if (subUp.fromNgsiv1) ->
* either setCondsAndInitialNotifyNgsiv1 or setCondsAndInitialNotify. 

But **if (subUp.fromNgsiv1)** is in the  **if (subUp.subjectProvided)**, which is excepted for NGSIV2 subscription. I am missing something?

Reply:
` subUp->subjectProvided      = (notifyConditionVector.size() > 0); `

I figured out, the setCondsAndInitialNotifyNgsiv1 is triggered when updating Context Subscription (v1) with a "notifyConditions" array.

Reply:
I'm writing specific functional tesst to trigger notifications when updating a subscription both on NGSIv1 and NGSV2. One question. How to get the subscriptionId from the response payload in the creation step, in order to use it in the update step (to create the /v2/subscription/subscriptionId url)?

Reply:
> NOTE: maybe this function should be renamed setCondsAndInitialNotifyNgsiv2() to make things more clear.

I would keep the name of the function as it is, because also the mongoSubscribeContext (V1) uses it (through mongoCreateSubscription)

Reply:
> Have a look for instance to the test named Subscription with duplicated items in attrs filter. The trick it to get SUB_ID just after creation request this way:

Thanks. How to get the subscriptionId from the response payload in the creation step, in the NGSIV1 case instead (v1/subscribeContext) ? 
I need to extract the subscriptionId field from a JSON like the following:
```
{
    "subscribeResponse": {
        "subscriptionId": "5bf28aad456d31fc96bd52dc",
        "duration": "PT24H"
    }
}
```

Reply:
Ah ok... you need it for NGSIv1 :)

In that case, you can have a look to test named "X-Auth-Token From Subscribe Request".

```
SUB_ID=$(echo "$_response" | grep subscriptionId | awk -F\" '{ print $4 }')
```

Reply:
I need for both :) thanks

Reply:
Fixed in 5845ebd